resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/PodTemplateConfigMapStep.scala

```diff
@@ -31,14 +31,16 @@ private[spark] class PodTemplateConfigMapStep(conf: KubernetesConf)
 
   private val hasTemplate = conf.contains(KUBERNETES_EXECUTOR_PODTEMPLATE_FILE)
 
+  private val configmapName = s"${conf.resourceNamePrefix}-$POD_TEMPLATE_CONFIGMAP"
```

Comment:
This makes the following status.
```
tpcds-py-eebc4474fa60b6f7-driver-conf-map      1      5s
tpcds-py-eebc4474fa60b6f7-podspec-configmap    1      5s
```

This is inconsistent because Driver is using `driver-conf-map` postfix and executor is using `podspec-configmap`.
```
resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala:    val configMapName = s"${conf.resourceNamePrefix}-driver-conf-map"
```

Let's use `exec-conf-map` post fix.

Reply:
Wel actually this configmap is not used by the executors but also by the driver. The driver needs this configmap to spawn executors with the right template. So I'll change the name to:`driver-podspec-configmap`. Let me know if that is fine.

Reply:
Could you share your result from `k get cm`? Does it looks reasonably consistent?

Reply:
Also, please update your PR description. I added **AFTER** paragraph at this PR description. You can fill in the final output there.

Reply:
During running the integration tests when watching the configmap it looks something like this:

```
aaece65ef82e4a30b7b7800aad600d4f   spark-test-app-aac9f37502b2ca55-driver-conf-map   1      0s
aaece65ef82e4a30b7b7800aad600d4f   spark-test-app-aac9f37502b2ca55-driver-podspec-conf-map   1      0
```

For me this looks nice because it's clear that the confimap is used by the driver. And when sorted alphabetically they are neatly together.

Reply:
Got it. Thank you, @stijndehaes .

Reply:
Nice!